frontend/syntax.asdl

```diff
@@ -309,21 +309,21 @@ module syntax
   | Pipeline(List[command] children, bool negated, List[int] stderr_indices)
   | AndOr(List[id] ops, List[command] children)
     # Part of for, while, until (but not if, case, ShFunction).  No redirects.
-  | DoGroup(List[command] children)
+  | DoGroup(Token left, Token right, List[command] children)
```

Comment:
How about putting `Token right` last?  That is how I did it for single_quoted, double_quoted, command_sub, etc.

(which we should really rename to SingleQuoted, DoubleQuoted, CommandSub, since they are real concrete types, not sum types)